src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -595,6 +595,29 @@ internal bool IsEntryPointCandidate
             get { return IsStatic && Name == WellKnownMemberNames.EntryPointMethodName; }
         }
 
+        internal bool HasAsyncMainReturnType(CSharpCompilation compilation)
```

Comment:
>internal bool HasAsyncMainReturnType(CSharpCompilation compilation) [](start = 8, length = 67)

This looks like a very specialized helper, I wouldn't put it into MethodSymbol. Consider putting it next to the consumer and making it private. Also, the name is somewhat confusing because the method doesn't care whether the method is async or not. It only checks the return type for Task. Consider renaming the method to reflect that.